components/modal/confirm.tsx

```diff
@@ -6,98 +6,112 @@ import Dialog, { ModalFuncProps } from './Modal';
+const IS_REACT_16 = !!ReactDOM.createPortal;
 
+const ConfirmDialog = (props: ConfirmDialogProps) => {
+  const { onCancel, onOk, close, zIndex, afterClose, visible } = props;
+  const iconType = props.iconType || 'question-circle';
+  const okType = props.okType || 'primary';
+  const prefixCls = props.prefixCls || 'ant-confirm';
+  // 默认为 true，保持向下兼容
+  const okCancel = ('okCancel' in props) ? props.okCancel! : true;
+  const width = props.width || 416;
+  const style = props.style || {};
   // 默认为 false，保持旧版默认行为
   const maskClosable = props.maskClosable === undefined ? false : props.maskClosable;
-
-  // 默认为 true，保持向下兼容
-  if (!('okCancel' in props)) {
-    props.okCancel = true;
-  }
-
   const runtimeLocale = getConfirmLocale();
+  const okText = props.okText ||
+    (okCancel ? runtimeLocale.okText : runtimeLocale.justOkText);
+  const cancelText = props.cancelText || runtimeLocale.cancelText;
 
-  props.okText = props.okText ||
-    (props.okCancel ? runtimeLocale.okText : runtimeLocale.justOkText);
-  props.cancelText = props.cancelText || runtimeLocale.cancelText;
-
-  function close(...args: any[]) {
-    const unmountResult = ReactDOM.unmountComponentAtNode(div);
-    if (unmountResult && div.parentNode) {
-      div.parentNode.removeChild(div);
-    }
-    const triggerCancel = args && args.length &&
-      args.some(param => param && param.triggerCancel);
-    if (props.onCancel && triggerCancel) {
-      props.onCancel(...args);
-    }
-  }
-
-  let body = (
+  const body = (
     <div className={`${prefixCls}-body`}>
-      <Icon type={props.iconType!} />
+      <Icon type={iconType!} />
       <span className={`${prefixCls}-title`}>{props.title}</span>
       <div className={`${prefixCls}-content`}>{props.content}</div>
     </div>
   );
 
-  let footer: React.ReactElement<any> | null = null;
-  if (props.okCancel) {
-    footer = (
-      <div className={`${prefixCls}-btns`}>
-        <ActionButton actionFn={props.onCancel} closeModal={close}>
-          {props.cancelText}
-        </ActionButton>
-        <ActionButton type={props.okType} actionFn={props.onOk} closeModal={close} autoFocus>
-          {props.okText}
-        </ActionButton>
-      </div>
-    );
-  } else {
-    footer = (
-      <div className={`${prefixCls}-btns`}>
-        <ActionButton type={props.okType} actionFn={props.onOk} closeModal={close} autoFocus>
-          {props.okText}
-        </ActionButton>
-      </div>
-    );
-  }
+  const footer = okCancel ? (
+    <div className={`${prefixCls}-btns`}>
+      <ActionButton actionFn={onCancel} closeModal={close}>
+        {cancelText}
+      </ActionButton>
+      <ActionButton type={okType} actionFn={onOk} closeModal={close} autoFocus>
+        {okText}
+      </ActionButton>
+    </div>
+  ) : (
+    <div className={`${prefixCls}-btns`}>
+      <ActionButton type={okType} actionFn={onOk} closeModal={close} autoFocus>
+        {okText}
+      </ActionButton>
+    </div>
+  );
```

Comment:
A bit nitty, but in my companies code we would deduplicate a bit of this and write this:

```js
const footer = (
  <div className={`${prefixCls}-btns`}>
    {okCancel && (
      <ActionButton actionFn={onCancel} closeModal={close}>
        {cancelText}
      </ActionButton>
    )}
    <ActionButton type={okType} actionFn={onOk} closeModal={close} autoFocus>
      {okText}
    </ActionButton>
  </div>
);
```

Reply:
You are right.